Public/Src/Cache/ContentStore/Distributed/Redis/RedisBlobAdapter.cs

```diff
@@ -55,10 +55,64 @@ public RedisBlobAdapter(RedisDatabaseAdapter redis, TimeSpan blobExpiryTime, lon
             _tracer = tracer;
         }
 
+        /// <nodoc />
+        public class PutBlobResult : BoolResult
+        {
+            /// <nodoc />
+            public ContentHash Hash { get; }
+
+            /// <nodoc />
+            public long BlobSize { get; }
+
+            /// <nodoc />
+            public bool AlreadyInRedis { get; }
+
+            /// <nodoc />
+            public long? NewCapacityInRedis { get; }
+
+            /// <nodoc />
+            public string RedisKey { get; }
+
+            /// <nodoc />
+            public PutBlobResult(ContentHash hash, long blobSize, bool alreadyInRedis = false, long? newCapacity = null, string redisKey = null)
+            {
+                Hash = hash;
+                BlobSize = blobSize;
+                AlreadyInRedis = alreadyInRedis;
+                NewCapacityInRedis = newCapacity;
+                RedisKey = redisKey;
+            }
+
+            /// <nodoc />
+            public PutBlobResult(ContentHash hash, long blobSize, string errorMessage)
+                : base(errorMessage)
+            {
+                Hash = hash;
+                BlobSize = blobSize;
+            }
+
+            /// <inheritdoc />
+            public override string ToString()
+            {
+                string baseResult = $"Hash=[{Hash.ToShortString()}], BlobSize=[{BlobSize}]";
+                if (Succeeded)
+                {
+                    if (AlreadyInRedis)
+                    {
+                        return $"{baseResult}. PutBlob skipped: the hash is already in Redis.";
+                    }
+
+                    return $"{baseResult}. Successfully reserved space in {RedisKey}. NewCapacity=[{NewCapacityInRedis}].";

```

Comment:
Maybe less words and more formatting for this stuff. E.g. "Result=[{Success}] Key=[{RedisKey}] NewCapacity=[{NewCapacityInRedis}]" here and "AlreadyInRedis=[true]" above.